### PR TITLE
services/regulated-assets-approval-server-demo: add Dockerfile and Makefile

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Add build files for deploying regulated-assets-approval-server-demo.

### Why

So we can have CD/CI.

### CODE REVIEW

**Review by leighmcculloch — APPROVED** [review]

File: services/regulated-assets-approval-server-demo/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.14-stretch as build
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply services/regulated-assets-approval-server-demo/docker/Dockerfile:1]: My bad. Thanks for spotting that 👀

File: services/regulated-assets-approval-server-demo/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+ADD . /src/regulated-assets-approval-server-demo
+WORKDIR /src/regulated-assets-approval-server-demo
+RUN go build -o /bin/regulated-assets-approval-server-demo ./services/regulated-assets-approval-server-demo
+
+
+FROM ubuntu:16.04
```
[the remark this answers is not in the document]
Comment (jacekn) [reply services/regulated-assets-approval-server-demo/docker/Dockerfile:8]: Yes definitely go with 20.04

**Review by jacekn — APPROVED** [review]
This looks great and consistent with other services.
It would be worth adding entry in the top level `Makefile` too but feel free to do it in a separate commit

File: services/regulated-assets-approval-server-demo/docker/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply services/regulated-assets-approval-server-demo/docker/Dockerfile]: https://github.com/stellar/go/pull/3461/commits/f16b259899d75242d0c485b8c7deaa353f5555c5

File: services/regulated-assets-approval-server-demo/docker/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply services/regulated-assets-approval-server-demo/docker/Dockerfile]: https://github.com/stellar/go/pull/3461/commits/94e30ab749b9d46c30826d3a3cfa9524b12c5fc8

**Comment by marcelosalloum** [comment]
> This looks great and consistent with other services.
> It would be worth adding entry in the top level `Makefile` too but feel free to do it in a separate commit

Done that in this commit: https://github.com/stellar/go/pull/3461/commits/eb2be953156b2c998f4137023c83e78ec0a5fd56

**Review by jacekn — APPROVED** [review]

**Pull request merged**